README.md

```diff
@@ -74,97 +79,92 @@ df.columns = ['x', 'y', 'z1']
 
 Project docs are published here: https://koalas.readthedocs.io
 
+
 ## Mailing List
 
 We use Google Groups for mailling list: https://groups.google.com/forum/#!forum/koalas-dev
 
 
 ## Development Guide
 
-### Environment Setup
+See [CONTRIBUTING.md](https://github.com/databricks/koalas/master/CONTRIBUTING.md).
 
-We recommend setting up a Conda environment for development:
-```bash
-conda create --name koalas-dev-env python=3.6
-conda activate koalas-dev-env
-conda install -c conda-forge pyspark=2.4
-conda install -c conda-forge --yes --file requirements-dev.txt
-pip install -e .  # installs koalas from current checkout
-```
 
-Once setup, make sure you switch to `koalas-dev-env` before development:
-```bash
-conda activate koalas-dev-env
-```
+## Design Principles
 
-### Running Tests
+This section outlines design princples guiding the Koalas project.
 
-There is a script `./dev/pytest` which is exactly same as `pytest` but with some default settings to run Koalas tests easily.
+### Be Pythonic
 
-To run all the tests, similar to our CI pipeline:
-```bash
-# Run all unittest and doctest
-./dev/pytest
-```
+Koalas targets Python data scientists. We want to stick to the convention that users are already familiar with as much as possible. Here are some examples:
 
-To run a specific test file:
-```bash
-# Run unittest
-./dev/pytest -k test_dataframe.py
+- Function names and parameters use snake_case, rather than CamelCase. This is different from PySpark's design. For example, Koalas has `to_pandas()`, whereas PySpark has `toPandas()` for converting a DataFrame into a pandas DataFrame. In limited cases, to maintain compatibility with Spark, we also provide Spark's variant as an alias.
 
-# Run doctest
-./dev/pytest -k series.py --doctest-modules databricks
-```
+- Koalas respects to the largest extent the conventions of the Python numerical ecosystem, and allows the use of numpy types, etc. that are supported by Spark.
 
-To run a specific doctest/unittest:
-```bash
-# Run unittest
-./dev/pytest -k "DataFrameTest and test_Dataframe"
+- Koalas docs' style and infrastructure simply follow rest of the PyData projects'.
 
-# Run doctest
-./dev/pytest -k DataFrame.corr --doctest-modules databricks
-```
+### Unify small data (pandas) API and big data (Spark) API, but pandas first
 
-Note that `-k` is used for simplicity although it takes an expression. You can use `--verbose` to check what to filter. See `pytest --help` for more details.
+The Koalas DataFrame is meant to provide the best of pandas and Spark under a single API, with easy and clear conversions between each API when necessary. When Spark and pandas have similar APIs with subtle differences, the principle is to honor the contract of the pandas API first.
 
-### Building Documentation
+There are 4 different classes of functions:
 
-To build documentation via Sphinx:
+ 1. Functions that are only found in Spark (`select`, `selectExpr`). These functions should also be available in Koalas.
```

Comment:
I still am not sure if we should allow those. `select` conflicts with Pandas's (https://github.com/pandas-dev/pandas/blob/88062f75dbca929ec082295c936edd07cc912dbf/pandas/core/generic.py#L3653-L3669):

```python
>>> pd.DataFrame([{'a': 1}]).select
<bound method NDFrame.select of    a
0  1>
```

`selectExpr` means we might face later if we should allow, for instance, `filter` should accept string expressions as well, which conflicts with Pandas's.

I was wondering if we, instead, say:

Spark APIs can be added only with strong reasons when they don't conflict with Pandas.

Reply:
There is indeed a select function, but this is fine because the expected input is different and it is deprecated, see the doc here:
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.select.html#pandas.DataFrame.select

It does not conflict with pandas in the sense that we can add the full pandas behaviour later. For now, like we do with functions that are partially implemented, we throw a notimplementederror for the pandas arguments.

Given the ubiquity of select in spark, I believe we should add it.

Reply:
I think at least we shouldn't pick `select` as an example in this doc. and .. if you won't mind, I would stick to "Spark APIs can be added only with strong reasons when they don't conflict with Pandas." ..